lib/taskana-core/src/main/java/pro/taskana/TaskanaEngineConfiguration.java

```diff
@@ -489,31 +283,415 @@ public static boolean shouldUseLowerCaseForAccessIds() {
     return true;
   }
 
-  public Properties readPropertiesFromFile() {
-    return loadProperties(this.propertiesFileName);
-  }
+  public static class TaskanaEngineConfigurationBuilder {
+
+    private static final Logger LOGGER =
+        LoggerFactory.getLogger(TaskanaEngineConfigurationBuilder.class);
+
+    private static final String TASKANA_PROPERTIES = "/taskana.properties";
+    private static final String TASKANA_PROPERTY_SEPARATOR = "|";
+
+    // TASKANA_SCHEMA_VERSION
+    private static final String DEFAULT_SCHEMA_NAME = "TASKANA";
 
-  private void initSchemaName(String schemaName) {
-    if (schemaName != null && !schemaName.isEmpty()) {
-      this.setSchemaName(schemaName);
-    } else {
-      this.setSchemaName(DEFAULT_SCHEMA_NAME);
+    private final DataSource dataSource;
+    private final boolean securityEnabled;
+    private final boolean useManagedTransactions;
+    private String schemaName;
+    private Properties properties;
+
+    private Map<TaskanaRole, Set<String>> roleMap = new HashMap<>();
```

Comment:
```suggestion
    private Map<TaskanaRole, Set<String>> roleMap = new EnumMap<>();
```
This is a Code Smell